src/algebraic_geometry/function_field.lean

```diff
@@ -73,4 +78,120 @@ lemma Scheme.germ_to_function_field_injective [is_integral X] (U : opens X.carri
   [nonempty U] : function.injective (X.germ_to_function_field U) :=
 germ_injective_of_is_integral _ _
 
+lemma generic_point_eq_of_is_open_immersion {X Y : Scheme} (f : X ⟶ Y) [H : is_open_immersion f]
+  [hX : irreducible_space X.carrier] [irreducible_space Y.carrier] :
+    f.1.base (generic_point X.carrier : _) = (generic_point Y.carrier : _) :=
+begin
+  apply ((generic_point_spec _).eq _).symm,
+  show t0_space Y.carrier, by apply_instance,
+  convert (generic_point_spec X.carrier).image (show continuous f.1.base, by continuity),
+  symmetry,
+  rw [eq_top_iff, set.top_eq_univ, set.top_eq_univ],
+  convert subset_closure_inter_of_is_preirreducible_of_is_open _ H.base_open.open_range _,
+  rw [set.univ_inter, set.image_univ],
+  apply_with preirreducible_space.is_preirreducible_univ { instances := ff },
+  show preirreducible_space Y.carrier, by apply_instance,
+  exact ⟨_, trivial, set.mem_range_self hX.2.some⟩,
+end
+
+noncomputable
+instance stalk_function_field_algebra [irreducible_space X.carrier] (x : X.carrier) :
+  algebra (X.presheaf.stalk x) X.function_field :=
+begin
+  apply ring_hom.to_algebra,
+  exact X.presheaf.stalk_specializes ((generic_point_spec X.carrier).specializes trivial)
+end
+
+instance function_field_is_scalar_tower [irreducible_space X.carrier] (U : opens X.carrier) (x : U)
+  [nonempty U] :
+  is_scalar_tower (X.presheaf.obj $ op U) (X.presheaf.stalk x) X.function_field :=
+begin
+  haveI : nonempty U := ⟨x⟩,
```

Comment:
`nonempty U` is also an assumption. Did you intend to remove that assumption?

Reply:
I intended to but failed since the type signature of the lemma needs this instance, and its quite hard to obtain `nonempty U` from `x` in the signature.